diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 0567685..b528a27 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -517,6 +517,8 @@ endif()
 if (BUILD_STATIC_LOADER)
     # TLDR: This feature only exists at the request of Google for Chromium. No other project should use this!
     message(NOTICE "Apple STATIC lib: it will be built but not installed, and vulkan.pc and VulkanLoaderConfig.cmake won't be generated!")
+    install(TARGETS vulkan EXPORT VulkanLoaderConfig)
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/vulkan.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig" RENAME "${API_TYPE}.pc")
     return()
 endif()
 
